exec/java-exec/src/main/java/org/apache/drill/exec/vector/BitVector.java

```diff
@@ -177,9 +167,16 @@ public boolean copyFromSafe(int inIndex, int outIndex, BitVector from) {
 
   @Override
   public void load(SerializedField metadata, DrillBuf buffer) {
-    assert this.field.matches(metadata);
-    int loaded = load(metadata.getValueCount(), buffer);
-    assert metadata.getBufferLength() == loaded;
+    assert field.matches(metadata);
```

Comment:
If you decide to add this back in the case I mentioned above where it was removed, it would probably be worth having them produce the same message when the assert fails.